src/main/java/io/vertx/ext/amqp/impl/AmqpConnectionImpl.java

```diff
@@ -294,6 +289,8 @@ public AmqpConnection createReceiver(String address, AmqpReceiverOptions receive
   private void configureTheSource(AmqpReceiverOptions receiverOptions, ProtonReceiver receiver) {
     org.apache.qpid.proton.amqp.messaging.Source source = (org.apache.qpid.proton.amqp.messaging.Source) receiver
       .getSource();
+
+    source.setCapabilities(receiverOptions.getCapabilities().stream().map(Symbol::valueOf).toArray(Symbol[]::new));
```

Comment:
This looks like it sets an empty array if there are no capabilities. It would be better not to call the setter in that case.

Reply:
Also, being able to change all this without any test failures probably indicates we need some tests.

Reply:
Because we update the tests accordingly. The CI didn't fail, but locally it has not been that peaceful.